vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -120,17 +120,30 @@ def fused_moe_kernel_gptq_awq(
     offs_token = tl.load(sorted_token_ids_ptr + offs_token_id)
     token_mask = offs_token < num_valid_tokens
 
+    off_experts = tl.load(expert_ids_ptr + pid_m).to(tl.int64)
+    if off_experts == -1:
+        # -----------------------------------------------------------
+        # Write back zeros to the output when the expert is not
+        # in the current EP rank.
+        accumulator = tl.zeros((BLOCK_SIZE_M, BLOCK_SIZE_N),
+                               dtype=compute_type)
+        offs_cn = pid_n * BLOCK_SIZE_N + tl.arange(0, BLOCK_SIZE_N)
+        c_ptrs = c_ptr + stride_cm * offs_token[:, None] + stride_cn * offs_cn[
+            None, :]
+        c_mask = token_mask[:, None] & (offs_cn[None, :] < N)
+        tl.store(c_ptrs, accumulator, mask=c_mask)
+        return
```

Comment:
Since this is duplicated, I think it'd be nice to factor this out into a helper function:
```
@triton.jit
def write_zeros_to_output(
    c_ptr, stride_cm, stride_cn,
    pid_n, N,
    offs_token, token_mask,
    BLOCK_SIZE_M, BLOCK_SIZE_N,
    compute_type
):
    accumulator = tl.zeros((BLOCK_SIZE_M, BLOCK_SIZE_N), dtype=compute_type)
    offs_cn = pid_n * BLOCK_SIZE_N + tl.arange(0, BLOCK_SIZE_N)
    c_ptrs = c_ptr + stride_cm * offs_token[:, None] + stride_cn * offs_cn[None, :]
    c_mask = token_mask[:, None] & (offs_cn[None, :] < N)
    tl.store(c_ptrs, accumulator, mask=c_mask)
```